bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -62,17 +71,34 @@ def kfcv_awkde(sample, bwchoice, alphachoice, k=2):
 
 
 def optimizedparam(sampleval, nfold=2):
-    # Test wide range of bandwidths in addition to standard values
-    bwgrid = ['scott', 'silverman'] + numpy.logspace(-2,0,10).tolist()
-    # Fix adaptive sensitivity parameter to 1
-    alphagrid = [1]
+    # bw-grid and alpha-grid are values given as arguments to the function
+    bwgrid = args.bw_grid
+    alphagrid = args.alpha_grid
     FOM = {}
     for gbw in bwgrid:
         for alphavals in alphagrid:
-            FOM[(gbw, alphavals)] = kfcv_awkde(sampleval, gbw, alphavals, k=nfold)
+            FOM[(gbw, alphavals)] = kfcv_awkde(sampleval, gbw, alphavals,\
+                k=nfold)
     optval = max(FOM.items(), key=operator.itemgetter(1))[0]
     optbw, optalpha = optval[0], optval[1]
     maxFOM = FOM[(optbw, optalpha)]
+
+    # Plotting FOM paramters
+    fig = plt.figure(figsize=(12,8))
+    ax = fig.add_subplot(111)
+    for bw in bwgrid:
+        FOMlist = [FOM[(bw, al)] for al in alphagrid]
+        ax.plot(alphagrid, FOMlist, label='{0:.3f}'.format(bw))
+    ax.plot(optalpha, maxFOM, 'ko', linewidth=10, label= \
+            r'$\alpha={0:.3f},bw= {1:.3f}$'.format(optalpha, optbw))
+    ax.set_xlabel(r'$\alpha$', fontsize=15)
+    ax.set_ylabel(r'$FOM$', fontsize=15)
+    ax.set_ylim(maxFOM-20, maxFOM+5)
+    handles, labels = ax.get_legend_handles_labels()
```

Comment:
is this needed? ie do you need to get these values? what happens if you do plt.legend() without using handles/labels?